### ISSUE TITLE
[inbound filters] Ability to combine conditions in custom inbound filters

### ISSUE THREAD

**Issue by GaurangTandon** [getsentry/sentry#71901]
### Problem Statement

I often need to discard future events for an issue quickly because it is spamming our Sentry instance, and causing us to go over our limited project quota.

However, I would prefer to keep existing events as they are, so I can refer to them in the future (when implementing a bug fix for that issue).

I cannot update my application's code to update the beforeSend callback - for example - because 1. the updated app can take a few days to rollout to 100% users, and 2. some users don't update their app even if we have rolled it out from our side. So, by the time my beforeSend callback update will deploy, our Sentry instance would have already exhausted its project quota.

### Solution Brainstorm

I need a way to discard future events for an issue, without deleting the existing events for that issue, from the Sentry Web UI. 

Currently, in Sentry, I can "delete existing events", or I can "delete and discard future events". Judging by that it looks like just "discard future events" (my request) should be possible to implement on the server.

### Product Area

Issues

**Comment by lobsterkatie** [comment]
Hi, @GaurangTandon - It sounds like [inbound filters](https://docs.sentry.io/concepts/data-management/filtering/) might be what you want.

**Comment by GaurangTandon** [comment]
Are you referring to this? @lobsterkatie Is there a way for me to combine release and version number in the inbound filter. For example: release matching "1.2.3" and error starting with "TypeError: "?

![image](https://github.com/getsentry/sentry/assets/6308683/bc081c27-68a2-49e7-bca0-85fe0df6e18d)

**Comment by lobsterkatie** [comment]
Yes (that is what I meant) but unfortunately no (you can't have compound filters). It's a good feature for us to consider, though. I'll ping our PMs and add it to our backlog.

**Comment by iker-barriocanal** [comment]
Relay supports compound filters that match on [(almost) any field](https://github.com/getsentry/relay/blob/4cf796912d4b53cfd7f682586813c795b3c0f323/relay-event-schema/src/protocol/event.rs#L643) in a payload through [`GenericFilter`s](https://github.com/getsentry/sentry/blob/7ee09273bf59e5f7279cb2e84a86d891fd7c4df9/src/sentry/relay/config/__init__.py#L201), so only the sentry part needs to be updated. I'm not sure who owns that part so I'm bouncing it back to issues, but feel free to reroute. I'm happy to help with the implementation of including the filters in global/project configs.

**Comment by lobsterkatie** [comment]
Awesome, thanks, @iker-barriocanal. Will keep that in mind for the future!
